leash.go

```diff
@@ -100,23 +109,39 @@ func run(options GlobalOptions) {
 
 		// start a goroutine that reads from responses and logs.
 		responses := libhoney.Responses()
-		go handleResponses(responses, toBeResent, delaySending, options)
+		stats := newResponseStats()
+		go handleResponses(responses, toBeResent, delaySending, stats, options)
 
 		parsersWG.Add(1)
-		go func(plines chan string) {
+		go func(pTailer *tail.Tailer) {
 			// ProcessLines won't return until lines is closed
-			parser.ProcessLines(plines, toBeSent)
+			parser.ProcessLines(pTailer.Lines, toBeSent)
 			// trigger the sending goroutine to finish up
 			close(toBeSent)
 			// wait for all the events in toBeSent to be handed to libhoney
 			<-doneSending
 			parsersWG.Done()
-		}(lines)
+		}(tailer)
+		hny := &hny{
+			parser: parser,
+			tailer: tailer,
+			rStats: stats,
+		}
+		hnys = append(hnys, hny)
+		go logStats(hny, options.StatusInterval)
```

Comment:
afaict, the purpose of the `hny` struct is basically because you want to do one last pass over everything at line 139. and you only need to do that because you want to do it one last time before everything shuts down, but you need access to these locally-defined things (`parser`, `tailer`, `stats`).

idea: instead of firing off `len(tailers)` number of `logStats` goroutines, each with their own set of 3 things to log out + their own ticker + their own early return, why not group all of the things that we want to log out and do all of them at once, in a ticker loop and on shutdown?

bonus: `tailer` and `parser` already have a `LogStats()` function defined, and this is the only invocation of `logAndReset` on `responseStats`, so you can rename *that* easily.

suggestion -- define a:

    type StatsLogger interface{
      LogStats()
    }

and instead of building up a `hny{}` and then a `[]hny{}`, just build up a `[]StatsLogger` for everything, then call `go logStats` *once*, outside the `for ... range tailers` loop, and once right before shutdown.

your instinct that the `hny` struct was weird was spot on. do less work.